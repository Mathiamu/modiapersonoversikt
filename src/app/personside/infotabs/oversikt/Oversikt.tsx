import * as React from 'react';
import styled from 'styled-components';
import Oversiktskomponent from './Oversiktskomponent';
import { pxToRem } from '../../../../styles/personOversiktTheme';
import VarselOversikt from './VarselOversikt';
import OppfolgingOversikt from './OppfolgingOversikt';
import YtelserOversikt from './YtelserOversikt';
import UtbetalingerOversikt from './UtbetalingerOversikt';
import MeldingerOversikt from './MeldingerOversikt';
import { INFOTABS } from '../InfoTabEnum';

const Style = styled.article`
    display: flex;
`;

const KolonneStyle = styled.div`
    > * {
        margin: ${pxToRem(15)};
        max-width: 30rem;
    }
`;

function Oversikt() {
    return (
        <Style>
            <KolonneStyle>
                <Oversiktskomponent
                    infotabToOpen={INFOTABS.UTBETALING}
                    tittel={'Utbetalinger'}
                    children={<UtbetalingerOversikt />}
                />
                <Oversiktskomponent
                    infotabToOpen={INFOTABS.OPPFOLGING}
                    tittel={'OppfÃ¸lging'}
                    children={<OppfolgingOversikt />}
                />
                <Oversiktskomponent infotabToOpen={INFOTABS.VARSEL} tittel={'Varsler'} children={<VarselOversikt />} />
            </KolonneStyle>
            <KolonneStyle>
                <Oversiktskomponent
                    infotabToOpen={INFOTABS.MELDINGER}
                    tittel={'Meldinger'}
                    children={<MeldingerOversikt />}
                />
                <Oversiktskomponent
                    infotabToOpen={INFOTABS.YTELSER}
                    tittel={'Ytelser'}
                    children={<YtelserOversikt />}
                />
            </KolonneStyle>
        </Style>
    );
}

export default Oversikt;
