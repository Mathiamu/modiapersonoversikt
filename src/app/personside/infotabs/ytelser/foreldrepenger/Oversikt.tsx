import * as React from 'react';
import { Foreldrepengerrettighet } from '../../../../../models/ytelse/foreldrepenger';
import theme from '../../../../../styles/personOversiktTheme';
import { utledMaksDato, utledFraDatoForRettighet } from './foreldrePengerUtils';
import { datoVerbose } from '../../utbetalinger/utils/utbetalingerUtils';
import styled from 'styled-components';
import DescriptionList from '../DescriptionList';
import YtelserBullet from '../YtelserBullet';

interface Props {
    foreldrePenger: Foreldrepengerrettighet;
}

const Wrapper = styled.div`
    margin: 0 ${theme.margin.px20} ${theme.margin.px20};
  > *:not(:first-child) {
    border-top: ${theme.border.skilleDashed};
  }
`;

function Oversikt({foreldrePenger}: Props) {

    const foreldrePengeRetten = {
        Foreldrepengetype: foreldrePenger.foreldrepengetype,
        Dekningsgrad: foreldrePenger.dekningsgrad + '%',
        Dagsats: 'Ikke implementert',
        Graderingsdager: 'Ikke implementert',
        'Fedrekvote tilogmed': 'Ikke implementert',
        'Mødrekvote tilogmed': 'Ikke implementert',
        'Rettighet fra dato': datoVerbose(utledFraDatoForRettighet(foreldrePenger)).sammensatt,
        Restdager: foreldrePenger.restDager,
        Maksdato: utledMaksDato(foreldrePenger),
        Arbeidskategori: 'Ikke implementert'
    };

    const barnet = {
        Termindato: foreldrePenger.rettighetFom && datoVerbose(foreldrePenger.rettighetFom).sammensatt,
        Fødselsdato: foreldrePenger.barnetsFødselsdato && datoVerbose(foreldrePenger.barnetsFødselsdato).sammensatt,
        'Annen forelder': foreldrePenger.andreForeldersFnr,
        Omsorgsovertakelse: 'Ikke implementert',
        'Foreldre av samme kjønn': 'Ikke implementert',
        'Antall barn': foreldrePenger.antallBarn
    };

    return (
        <Wrapper>
            <YtelserBullet tittel="Om foreldrepengeretten">
                <DescriptionList entries={foreldrePengeRetten}/>
            </YtelserBullet>
            <YtelserBullet tittel="Om barnet">
                <DescriptionList entries={barnet}/>
            </YtelserBullet>
        </Wrapper>
    );
}

export default Oversikt;
