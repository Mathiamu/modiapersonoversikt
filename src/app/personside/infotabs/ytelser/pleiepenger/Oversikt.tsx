import * as React from 'react';
import { Pleiepengerettighet } from '../../../../../models/ytelse/pleiepenger';
import { OversiktStyling } from '../felles-styling/CommonStylingYtelser';
import YtelserBullet from '../felles-styling/YtelserBullet';
import DescriptionList from '../felles-styling/DescriptionList';
import ForbrukteDager from './ForbrukteDager';

interface Props {
    pleiepenger: Pleiepengerettighet;
}

function Oversikt({pleiepenger}: Props) {

    const omPleiepengerettenEntries = {
        'Fra og med': 'Ikke implementert',
        'Til og med': 'Ikke implementert',
        Kompensasjonsgrad: 'Ikke implementert',
        Pleiepengegrad: 'Ikke implementert',
        'Totalt invilget': 'Ikke implementert',
        Ferieperioder: 'Ikke implementert',
        Barnet: pleiepenger.barnet,
        'Annen forelder': pleiepenger.andreOmsorgsperson
    };

    const arbeidsSitsuasjonEntries = {
        Arbeidsgiver: '',
        Arbeidskategori: '',
        Inntekstsperiode: '',
        Kontonummer: '',
        Refusjonstype: '',
        'Inntekt for perioden': '',
        'Refusjon til dato': ''
    };

    return (
        <OversiktStyling>
            <YtelserBullet tittel="Om pleiepengeretten">
                <ForbrukteDager pleiepenger={pleiepenger}/>
                <DescriptionList entries={omPleiepengerettenEntries}/>
            </YtelserBullet>
            <YtelserBullet tittel="Arbeidssituasjon">
                <DescriptionList entries={arbeidsSitsuasjonEntries}/>
            </YtelserBullet>
        </OversiktStyling>);
}

export default Oversikt;
