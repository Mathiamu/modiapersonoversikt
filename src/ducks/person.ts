import { Person } from '../models/person';
import { Action, Dispatch } from 'redux';
import { FetchError, FetchSuccess, RestActions, STATUS } from './utils';

export interface PersonState {
    data: Person;
    status: STATUS;
    error: string | null;
}

const mockPerson = {
    fornavn: 'Aremark',
    etternavn: 'Testfamilien',
    fodselsnummer: '10108000398'
};

const reducerName = 'person/';

const actionNames = {
    PENDING: reducerName + STATUS.PENDING,
    OK: reducerName + STATUS.OK,
    ERROR: reducerName + STATUS.ERROR
};

const initialState: PersonState = {
    data: mockPerson,
    status: STATUS.NOT_STARTED,
    error: null
};

function mockFetch(fodselsnummer: string) {
    return new Promise((resolve, reject) => {
        setTimeout(() => { resolve(mockPerson); }, 2500);
    });
}

export function hentPerson(fodselsnummer: string) {
    return (dispatch: Dispatch<Action>) => {
        dispatch({ type: actionNames.PENDING});
        return mockFetch(fodselsnummer)
            .then((person) => {
                dispatch({type: actionNames.OK, data: person});
                return person;
            })
            .catch((err) => {
                dispatch({type: actionNames.ERROR, error: err});
            });
    };
}

export default function reducer(state: PersonState = initialState, action: RestActions<Person>): PersonState {
    switch (action.type) {
        case actionNames.PENDING:
            return {...state, status: STATUS.PENDING };
        case actionNames.OK:
            return {...state, status: STATUS.OK, data: (<FetchSuccess<Person>> action).data};
        case actionNames.ERROR:
            return {...state, status: STATUS.ERROR, error: (<FetchError> action).error};
        default:
            return state;
    }
}