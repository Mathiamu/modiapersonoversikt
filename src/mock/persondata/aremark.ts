import {
    AdresseBeskyttelse,
    EgenAnsatt,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    FullmaktsRolle,
    Kjonn,
    LocalDate,
    LocalDateTime,
    Person,
    PersonStatus,
    SikkerhetstiltakType,
    SivilstandType,
    Skifteform
} from '../../app/personside/visittkort-v2/PersondataDomain';

export const barnAremark: ForelderBarnRelasjon[] = [
    {
        ident: '12345678910',
        rolle: ForelderBarnRelasjonRolle.BARN,
        navn: [
            {
                fornavn: 'BARN1',
                mellomnavn: null,
                etternavn: 'BARNESEN'
            }
        ],
        fodselsdato: ['2008-03-15' as LocalDate],
        alder: 13,
        adressebeskyttelse: [
            {
                kode: AdresseBeskyttelse.UGRADERT,
                beskrivelse: 'UGRADERT'
            }
        ],
        bostedAdresse: [
            {
                linje1: 'Islandsgate 49',
                linje2: '7693 Hovedøya',
                linje3: 'Norge'
            }
        ],
        personstatus: [
            {
                kode: PersonStatus.BOSATT,
                beskrivelse: 'BOSATT'
            }
        ]
    }
];

export const aremark: Person = {
    fnr: '10108000398',
    navn: [
        {
            fornavn: 'AREMARK',
            mellomnavn: null,
            etternavn: 'TESTFAMILIEN'
        }
    ],
    kjonn: [
        {
            kode: Kjonn.M,
            beskrivelse: 'Mann'
        }
    ],
    fodselsdato: ['1980-10-10' as LocalDate],
    alder: 41,
    dodsdato: [],
    bostedAdresse: [
        {
            linje1: 'Islandsgate 49',
            linje2: '7693 Hovedøya',
            linje3: 'Norge'
        }
    ],
    kontaktAdresse: [
        {
            linje1: 'Aremarkveien 1',
            linje2: '0320 Aremark',
            linje3: null
        }
    ],
    navEnhet: {
        id: '0219',
        navn: 'NAV Bærum'
    },
    statsborgerskap: [
        {
            land: {
                kode: 'NOR',
                beskrivelse: 'NORGE'
            },
            gyldigFraOgMed: '1980-10-10' as LocalDate,
            gyldigTilOgMed: null,
            erHistorisk: false
        }
    ],
    adressebeskyttelse: [
        {
            kode: AdresseBeskyttelse.KODE6,
            beskrivelse: 'Ingen'
        }
    ],
    sikkerhetstiltak: [
        {
            type: SikkerhetstiltakType.TFUS,
            gyldigFraOgMed: '2005-02-13' as LocalDate,
            gyldigTilOgMed: '2030-02-15' as LocalDate
        }
    ],
    erEgenAnsatt: EgenAnsatt.JA,
    personstatus: [
        {
            kode: PersonStatus.BOSATT,
            beskrivelse: 'Bosatt i Norge'
        }
    ],
    sivilstand: [
        {
            type: {
                kode: SivilstandType.GIFT,
                beskrivelse: 'Gift'
            },
            gyldigFraOgMed: '2005-12-12' as LocalDate
        }
    ],
    foreldreansvar: [
        {
            ansvar: 'felles',
            ansvarlig: {
                fornavn: 'Test',
                etternavn: 'Testesen',
                mellomnavn: null
            },
            ansvarsubject: {
                fornavn: 'Barn',
                etternavn: 'Barnesen',
                mellomnavn: null
            }
        }
    ],
    deltBosted: [
        {
            startdatoForKontrakt: '2000-10-10' as LocalDate,
            sluttdatoForKontrakt: '2005-05-07' as LocalDate,
            adresse: {
                linje1: `Adresseveien 1`,
                linje2: '0000 Aremark',
                linje3: null
            }
        }
    ],
    dodsbo: [
        {
            adressat: {
                advokatSomAdressat: {
                    kontaktperson: {
                        fornavn: 'Advokat',
                        mellomnavn: null,
                        etternavn: 'Navnesen'
                    },
                    organisasjonsnavn: 'Advokatkontoret Aremark',
                    organisasjonsnummer: '01234567891011'
                },
                organisasjonSomAdressat: null,
                personSomAdressat: null
            },
            adresse: {
                linje1: 'Elgelia 20, 0000 Aremark',
                linje2: null,
                linje3: null
            },
            registrert: '2010-02-02' as LocalDate,
            skifteform: Skifteform.OFFENTLIG
        }
    ],
    fullmakt: [
        {
            motpartsPersonident: '123456789',
            motpartsPersonNavn: {
                fornavn: 'Navn',
                mellomnavn: null,
                etternavn: 'Navnesen'
            },
            motpartsRolle: FullmaktsRolle.FULLMEKTIG,
            omraade: ['*'],
            gyldigFraOgMed: '2015-01-01' as LocalDate,
            gyldigTilOgMed: '2017-12-12' as LocalDate
        }
    ],
    vergemal: [
        {
            ident: '21042900076',
            navn: {
                fornavn: 'Simen',
                mellomnavn: null,
                etternavn: 'Solli'
            },
            vergesakstype: 'voksen',
            omfang: 'personligeOgOekonomiskeInteresser',
            embete: 'Fylkesmannen i Troms og Finnmark',
            gyldighetstidspunkt: '2016-03-27' as LocalDate,
            opphoerstidspunkt: '2017-06-30' as LocalDate
        }
    ],
    tilrettelagtKommunikasjon: {
        talesprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ],
        tegnsprak: [
            {
                kode: 'NO',
                beskrivelse: 'Norsk'
            }
        ]
    },
    telefonnummer: [
        {
            retningsnummer: {
                kode: '+47',
                beskrivelse: 'retningsnummer er norsk'
            },
            identifikator: '99009900',
            sistEndret: '2017-10-10T10:15:30' as LocalDateTime,
            sistEndretAv: 'TESTFAMILIEN AREMARK',
            prioritet: 1
        }
    ],
    kontaktOgReservasjon: {
        personident: '10108000398',
        reservasjon: 'true',
        epostadresse: {
            value: 'epost@nav.no',
            sistOppdatert: null,
            sistVerifisert: '2013-01-01' as LocalDate
        },
        mobiltelefonnummer: {
            value: '90000000',
            sistOppdatert: null,
            sistVerifisert: null
        }
    },
    bankkonto: {
        kontonummer: '1234567890',
        banknavn: 'DNB ASA',
        sistEndret: '2006-03-15T00:00:00' as LocalDateTime,
        sistEndretAv: '1010800 BD03',

        bankkode: null,
        swift: 'DNBANOKKXXX',
        landkode: null,
        adresse: {
            linje1: 'Bankveien 1,',
            linje2: '0357 Bankestad',
            linje3: null
        },
        valuta: {
            kode: 'NOK',
            beskrivelse: 'Norske kroner'
        }
    },
    forelderBarnRelasjon: barnAremark
};
