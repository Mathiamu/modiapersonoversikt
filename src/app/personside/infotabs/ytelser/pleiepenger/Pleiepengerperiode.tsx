import * as React from 'react';
import { Pleiepengeperiode } from '../../../../../models/ytelse/pleiepenger';
import theme from '../../../../../styles/personOversiktTheme';
import styled from 'styled-components';
import YtelserPeriode from '../felles-styling/YtelserPeriode';
import { Undertittel } from 'nav-frontend-typografi';
import { formaterDato, NOKellerNull, prosentEllerNull } from '../../../../../utils/stringFormatting';
import { createTable } from '../../../../../utils/tableUtils';

interface Props {
    periode: Pleiepengeperiode;
    periodeNummer: number;
}

const VedtaksTable = styled.div`
    padding: 1rem;
    > *:not(:last-child) {
        margin-bottom: 1rem;
    }
    ${theme.table}
`;

function Pleiepengerperiode({ periode, ...props }: Props) {
    const tittelRekke = ['Fra og med', 'Til og med', 'Bruttobeløp', 'Anvist utbetaling', 'Dagsats', 'Pleiepengegrad'];
    const rows = periode.vedtak.map(vedtak => [
        formaterDato(vedtak.periode.fom),
        formaterDato(vedtak.periode.tom),
        NOKellerNull(vedtak.bruttobeløp),
        formaterDato(vedtak.anvistUtbetaling),
        NOKellerNull(vedtak.dagsats),
        prosentEllerNull(vedtak.pleiepengegrad)
    ]);
    const table = createTable(tittelRekke, rows);

    return (
        <YtelserPeriode tittel={`Periode ${props.periodeNummer} - ${formaterDato(periode.fom)}`}>
            <VedtaksTable>
                <Undertittel>Anviste utbetalinger</Undertittel>
                {table}
            </VedtaksTable>
        </YtelserPeriode>
    );
}

export default Pleiepengerperiode;
