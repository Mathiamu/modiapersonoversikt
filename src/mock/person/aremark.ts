import { Diskresjonskoder } from '../../constants';
import { Person } from '../../models/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: 'M',
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    status: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: 'GIFT',
        beskrivelse: 'Gift'
    },
    familierelasjoner: []
};