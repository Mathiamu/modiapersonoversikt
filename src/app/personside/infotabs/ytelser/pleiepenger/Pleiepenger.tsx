import * as React from 'react';
import { Pleiepengerettighet } from '../../../../../models/ytelse/pleiepenger';
import Pleiepengerperiode from './Pleiepengerperiode';
import Oversikt from './Oversikt';
import { datoStigende } from '../../../../../utils/dateUtils';
import VisuallyHiddenAutoFokusHeader from '../../../../../components/VisuallyHiddenAutoFokusHeader';
import { erModiabrukerdialog } from '../../../../../utils/erNyPersonoversikt';
import { useOnMount } from '../../../../../utils/customHooks';
import { loggEvent } from '../../../../../utils/frontendLogger';
import styled from 'styled-components';
import theme from '../../../../../styles/personOversiktTheme';
import { useRef } from 'react';
import { guid } from 'nav-frontend-js-utils';

interface Props {
    pleiepenger: Pleiepengerettighet;
}

const StyledArticle = styled.article`
    ${theme.hvittPanel};
`;

function Pleiepenger(props: Props) {
    const titleId = useRef(guid());
    useOnMount(() => {
        loggEvent('Visning', 'Pleiepenger');
    });

    const sortertePerioder = props.pleiepenger.perioder.sort(datoStigende(p => p.fom)).reverse();

    return (
        <StyledArticle aria-labelledby={titleId.current}>
            <h2 className="sr-only" id={titleId.current}>
                Pleiepenger
            </h2>
            {erModiabrukerdialog() && <VisuallyHiddenAutoFokusHeader tittel="Pleiepengerettighet" />}
            <Oversikt pleiepenger={props.pleiepenger} />
            <ol>
                {sortertePerioder.map((periode, index) => (
                    <Pleiepengerperiode periodeNummer={sortertePerioder.length - index} key={index} periode={periode} />
                ))}
            </ol>
        </StyledArticle>
    );
}

export default Pleiepenger;
