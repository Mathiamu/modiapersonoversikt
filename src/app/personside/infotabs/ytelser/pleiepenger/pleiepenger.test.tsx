import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Oversikt from './Oversikt';
import { getMockPleiepengerettighet } from '../../../../../mock/ytelse/pleiepenger-mock';
import { aremark } from '../../../../../mock/person/aremark';
import TestProvider from '../../../../../test/Testprovider';
import Pleiepenger from './Pleiepenger';
import Pleiepengerperiode from './Pleiepengerperiode';

const mockPleiepenger = getMockPleiepengerettighet(aremark.fÃ¸dselsnummer);

Date.prototype.getTime = jest.fn(() => 0);

test('Om pleiepengeretten matcher snapshot', () => {
    const resultat = renderer.create(
        <TestProvider><Pleiepenger pleiepenger={mockPleiepenger}/></TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});

test('Om Oversikten i pleiepengeretten matcher snapshot', () => {
    const resultat = renderer.create(
        <TestProvider><Oversikt pleiepenger={mockPleiepenger}/></TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});

test('Om Pleiepengeperiode i pleiepengeretten matcher snapshot', () => {
    const resultat = renderer.create(
        <TestProvider>
            <Pleiepengerperiode periode={mockPleiepenger.perioder[0]} periodeNummer={1}/>
        </TestProvider>
    );

    expect(resultat.toJSON()).toMatchSnapshot();
});
