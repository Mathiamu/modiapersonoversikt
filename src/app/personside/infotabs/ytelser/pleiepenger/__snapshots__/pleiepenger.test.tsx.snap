// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om Oversikten i pleiepengeretten matcher snapshot 1`] = `
.c1 {
  border-radius: 0.5rem;
  background-color: #e9e7e7;
  box-shadow: inset 0 0 0 0.3rem white;
  padding: 1.25rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 dt {
  color: #78706a;
}

.c2 dd {
  font-weight: bold;
  margin-top: 0.3rem;
}

.c2 > div {
  margin: 0.9375rem 0;
  padding-right: 1rem;
  width: 13rem;
  overflow-wrap: break-word;
}

.c4 span:not(:last-child):after {
  content: '.';
}

.c3 li:not(:first-child) {
  border-top: solid 0.0625rem #c6c2bf;
}

.c0 {
  margin: .8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h3
      className="typo-ingress"
    >
      Om pleiepengeretten
    </h3>
    <dl
      className="c2"
    >
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Fra og med
        </dt>
        <dd
          className="typo-normal"
        >
          01.01.1970
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Til og med
        </dt>
        <dd
          className="typo-normal"
        >
          01.01.1970
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Pleiepengegrad
        </dt>
        <dd
          className="typo-normal"
        >
          66%
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Barnet (jente)
        </dt>
        <dd
          className="typo-normal"
        >
          08031815240
        </dd>
      </div>
      <div>
        <dt
          className="typo-etikett-liten"
        >
          Annen forelder
        </dt>
        <dd
          className="typo-normal"
        >
          26087625779
        </dd>
      </div>
    </dl>
  </div>
  <div
    className="c1"
  >
    <h3
      className="typo-ingress"
    >
      Arbeidssituasjon
    </h3>
    <ol
      aria-label="Arbeidsgivere"
      className="c3"
    >
      <li>
        <dl
          className="c2"
        >
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Arbeidsgiver
            </dt>
            <dd
              className="typo-normal"
            >
              Ryan og Sønner
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Arbeidskategori
            </dt>
            <dd
              className="typo-normal"
            >
              Arbeidstaker
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Inntekstsperiode
            </dt>
            <dd
              className="typo-normal"
            >
              Månedssats
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Kontonummer
            </dt>
            <dd
              className="typo-normal"
            >
              <span
                className="c4"
              >
                <span>
                  5493
                </span>
                <span>
                  85
                </span>
                <span>
                  21445
                </span>
              </span>
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Refusjonstype
            </dt>
            <dd
              className="typo-normal"
            >
              Ikke refusjon
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Inntekt for perioden
            </dt>
            <dd
              className="typo-normal"
            >
              45,029.00 NOK
            </dd>
          </div>
          <div>
            <dt
              className="typo-etikett-liten"
            >
              Refusjon til dato
            </dt>
            <dd
              className="typo-normal"
            >
              01.01.1970
            </dd>
          </div>
        </dl>
      </li>
    </ol>
    <button
      className="knapp knapp--hoved"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Vis alle arbeidsforhold
    </button>
  </div>
</div>
`;

exports[`Om Pleiepengeperiode i pleiepengeretten matcher snapshot 1`] = `
.c2 {
  text-transform: uppercase;
}

.c1 {
  font-weight: bold;
}

.c0 > *:first-child {
  background-color: #cce1f3;
  padding: 0.2rem 1.25rem;
}

.c0 ol {
  padding: 0;
  margin: 0;
}

.c4 {
  overflow: auto;
  box-shadow: 0 0 0 0.1rem rgba(0,0,0,0.25);
  border-radius: .25rem;
  background-color: white;
}

.c4 table {
  text-align: left;
  width: 100%;
}

.c4 table thead {
  border-bottom: 0.2rem solid rgba(0,0,0,0.15);
  font-weight: bold;
}

.c4 table th,
.c4 table td {
  padding: 0.7rem;
  vertical-align: middle;
}

.c4 table th {
  padding-bottom: 0.7rem;
}

.c4 table tr:nth-child(even) {
  background-color: rgba(0,0,0,0.1);
}

.c3 {
  padding: 1rem;
}

.c3 > *:not(:last-child) {
  margin-bottom: 1rem;
}

<li
  className="c0"
>
  <h3
    className="typo-normal"
  >
    <span
      className="c1"
    >
      <span
        className="c2"
      >
        Periode 1 - 01.01.1970
      </span>
    </span>
  </h3>
  <div
    className="c3"
  >
    <h2
      className="typo-undertittel"
    >
      Anviste utbetalinger
    </h2>
    <div
      className="typo-normal c4"
    >
      <table
        role="table"
      >
        <thead
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <th
              role="columnheader"
            >
              Fra og med
            </th>
            <th
              role="columnheader"
            >
              Til og med
            </th>
            <th
              role="columnheader"
            >
              Bruttobeløp
            </th>
            <th
              role="columnheader"
            >
              Anvist utbetaling
            </th>
            <th
              role="columnheader"
            >
              Dagsats
            </th>
            <th
              role="columnheader"
            >
              Pleiepengegrad
            </th>
          </tr>
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              120.00 NOK
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              5.00 NOK
            </td>
            <td
              role="cell"
            >
              66%
            </td>
          </tr>
          <tr
            role="row"
          >
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              596.00 NOK
            </td>
            <td
              role="cell"
            >
              01.01.1970
            </td>
            <td
              role="cell"
            >
              27.00 NOK
            </td>
            <td
              role="cell"
            >
              58%
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</li>
`;
